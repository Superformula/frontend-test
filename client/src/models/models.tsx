export class DdItem { // class model for Dropdown menu items
  content: string;
  selected: boolean;
  
  constructor(
    content: string,
    selected: boolean,
  ) {
    this.content = content;
    this.selected = selected;
  }
}

export class Review { // class model for Reviews
  revId: string; // Unique review id
  resId: string; // Id of restaurant that a given Review belongs to
  name: string; 
  date: string;
  rating: number;
  review: string;
  image: string;

  constructor(
    revId: string,
    resId: string,
    name: string,
    date: string,
    rating: number,
    review: string,
    image: string,
  ) {
    this.revId = revId;
    this.resId = resId;
    this.name = name;
    this.date = date;
    this.rating = rating;
    this.review = review;
    this.image = image
  }
}

export class Restaurant { // class model for Restaurants
  resId: string;
  title: string;
  category: string[];
  isOpen: boolean;
  cost: string;
  avg_rating: number;
  img: string;
  coordinates: number[];
  totReviews: number;
  reviews: Review[];
  location: string;
  constructor(
    resId: string,
    title: string,
    category: string[],
    isOpen: boolean,
    cost: string,
    avg_rating: number,
    img: string,
    totReviews: number,
    location: string,
    coordinates: number[],
    reviews: Review[],
  ) {
    this.resId = resId;
    this.title = title;
    this.category = category;
    this.isOpen = isOpen;
    this.cost = cost;
    this.avg_rating = avg_rating;
    this.img = img;
    this.coordinates = coordinates;
    this.totReviews = totReviews;
    this.reviews = reviews;
    this.location = location;
  }
}

