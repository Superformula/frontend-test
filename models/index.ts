export * from './Business';
export * from './Category';
export * from './Review';
