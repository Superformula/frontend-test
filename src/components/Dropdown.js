import * as React from 'react';
import onClickOutside from 'react-onclickoutside';
import CheckItem from './CheckItem';

class DropDownMenu extends React.Component {
    state = { isOpen: false }

    handleClickOutside = () => {
        const { isOpen } = this.state;
        // need this condition because otherwise
        // all dropdowns will open and close at the same time
        if (isOpen) {
            this.setState({ isOpen: false });
        }
        // also would have used functional component with
        // useState hook but was running into similiar problems there.
    }

    render() {
        const { listItems = [], menuTitle = '' } = this.props;
        const { isOpen } = this.state;
        return (
            <div className="dropdown_wrapper">
                <div className="dropdown_button" onClick={() => this.setState({ isOpen: !isOpen })}>
                    <span className="isOpen_text">{menuTitle}</span>
                    <span className="isOpen_text"><ion-icon name={isOpen ? 'arrow-dropup' : 'arrow-dropdown'}></ion-icon></span>
                </div>
                {isOpen && (
                    <div className="dropdown_list">
                        {listItems.map(item => (
                            <div key={item} className="dropdown_listitem">
                                <CheckItem name={item} />
                            </div>
                        ))}
                    </div>
                )}
            </div>
        );
    }
}

export default onClickOutside(DropDownMenu);