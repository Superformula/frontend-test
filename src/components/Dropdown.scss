// .dropdown-header {
// }
// .dropdown-header-title {
// }
// .dropdown-chevron {
// }
// .dropdown-item-list {
// }

// .dropdown-list-item {
// }

// .dropdown-wrapper {
// }
