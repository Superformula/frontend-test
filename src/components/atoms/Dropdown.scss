$max_dropdown_height: 250px;
$item_list_border_size: 1px;

$small_width: 80px;
$large_width: 230px;

$chevron_size: 4px;
$chevron_up_translation: calc(#{$chevron_size}/2);

.dropdown-wrapper {
    display: inline-block;
}

.dropdown-header {
    padding-top: 7px;
    padding-bottom: 7px;
    display: flex;
    justify-content: space-between;
    border-bottom: 1px solid #BCBCBC;
    background-color: white;

    &.small{
        width: calc(#{$small_width} + #{$item_list_border_size});
    }

    &.large{
        width: calc(#{$large_width} + #{$item_list_border_size});
    }
}

.dropdown-header-title {
    color: #021F44;
}

.dropdown-chevron-wrapper {
    margin-top: 7px;
    margin-bottom: 7px;
    margin-right: 3px;
};

.dropdown-chevron {
    border-left: 1px solid #A4A4A4;
    border-bottom: 1px solid #A4A4A4;
    width: $chevron_size;
    height: $chevron_size;

    &.chevron-up {
        // transform: rotateY(0deg) rotate(135deg) translateY(calc(-#{$chevron_size}/2);
        transform: translateY($chevron_up_translation) rotate(135deg);
    }

    &.chevron-down {
        transform: rotate(-45deg);
    }
}

.dropdown-item-list {
    background-color: white;
    border: $item_list_border_size solid #BCBCBC;
    position: absolute;
    overflow-y: auto;
    max-height: $max_dropdown_height;
    box-shadow: 0px 4px 4px #BCBCBC;
    transform: translateY(-1px);

    &.small{
        width: $small_width;
    }

    &.large{
        width: $large_width;
    }
}

.dropdown-list-item {
    color: #4D4D4D;
    padding: 5px;
}

.dropdown-list-item:hover {
    background-color: lightgray;
}

.dropdown-list-item:active {
    background-color: darkgray;
}
