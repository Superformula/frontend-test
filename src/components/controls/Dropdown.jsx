import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import { withStyles } from '@material-ui/core/styles';
import Menu from '@material-ui/core/Menu';
import MenuItem from '@material-ui/core/MenuItem';
import ListItemIcon from '@material-ui/core/ListItemIcon';
import CheckCircleIcon from '@material-ui/icons/CheckCircle';
import RadioButtonUncheckedIcon from '@material-ui/icons/RadioButtonUnchecked';
import { controlContainer } from './Controls.module.scss';
import { arrow, arrowUp } from './Dropdown.module.scss';
const StyledMenu = withStyles({
    paper: {
        border: '1px solid #d3d4d5',
        borderRadius: '0px',
        marginTop: '5px'
    }
})(props => (
    <Menu
        elevation={0}
        getContentAnchorEl={null}
        anchorOrigin={{
            vertical: 'bottom',
            horizontal: 'left'
        }}
        transformOrigin={{
            vertical: 'top',
            horizontal: 'left'
        }}
        {...props}
    />
));

const StyledMenuItem = withStyles(theme => ({
    root: {
        fontFamily: 'Helvetica Neue'
    }
}))(MenuItem);

const StyledIcon = withStyles(theme => ({
    root: {
        minWidth: '10px'
    }
}))(ListItemIcon);

const Dropdown = ({ label, items, selected, onChange, width }) => {
    const [anchorEl, setAnchorEl] = React.useState(null);

    function handleClick(event) {
        setAnchorEl(event.currentTarget);
    }

    function handleClose() {
        setAnchorEl(null);
    }

    return (
        <div className={controlContainer} style={{ width: `${width}px` }}>
            <div onClick={handleClick}>
                {label} <div className={classNames(arrow, { [arrowUp]: Boolean(anchorEl) })} />
            </div>
            <StyledMenu
                id="customized-menu"
                anchorEl={anchorEl}
                keepMounted
                open={Boolean(anchorEl)}
                onClose={handleClose}
            >
                {items.map(item => (
                    <StyledMenuItem
                        key={item.value}
                        onClick={() => {
                            onChange(item.value);
                            handleClose();
                        }}
                    >
                        <StyledIcon>
                            {selected === item.value ? (
                                <CheckCircleIcon size="small" />
                            ) : (
                                <RadioButtonUncheckedIcon style={{ fontSize: 18, color: 'black', minWidth: '30px' }} />
                            )}
                        </StyledIcon>
                        <div>{item.label}</div>
                    </StyledMenuItem>
                ))}
            </StyledMenu>
        </div>
    );
};

Dropdown.propTypes = {
    label: PropTypes.string,
    items: PropTypes.array,
    selected: PropTypes.string,
    onChange: PropTypes.func,
    width: PropTypes.number
};
export default Dropdown;
