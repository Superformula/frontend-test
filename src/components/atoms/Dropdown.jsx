import React from 'react';
import { PropTypes } from 'prop-types';
import onClickOutside from 'react-onclickoutside';
import combineClasses from '../../util/CombineClasses';
import './Dropdown.scss';

class Dropdown extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            isOpen: false
        };
    }

    toggleDropdown = () => {
        this.setState(prevState => {
            return {
                isOpen: !prevState.isOpen
            }
        });
    }

    collapseDropdown = () => {
        if (this.state.isOpen)
            this.setState({ isOpen: false });
    }

    handleClickOutside = () => {
        this.collapseDropdown();
    };

    renderChevron = () => {
        const chevronClassName = combineClasses([
            'dropdown-chevron',
            (this.state.isOpen ? 'chevron-up' : 'chevron-down')
        ]);

        return (
            <div className={ 'dropdown-chevron-wrapper' }>
                <div className={ chevronClassName } />
            </div>
        );
    }

    getSizeClass = () => {
        if (this.props.sizeSmall) return 'small';
        return 'large'; // default
    }

    renderHeader = () => {
        const headerClassName = combineClasses(['dropdown-header', this.getSizeClass()]);

        return (
            <div className={ headerClassName } onClick={ this.toggleDropdown }>
                <div className='dropdown-header-title'>{ this.props.title }</div>
                { this.renderChevron() }
            </div>
        );
    };

    renderItemList = () => {
        const itemListClassName = combineClasses([
            'dropdown-item-list',
            this.getSizeClass(),
            (this.state.isOpen ? 'dropdown-open' : 'dropdown-collapsed')
        ]);

        return (
            <div className={ itemListClassName }>
                { this.props.items.map(this.renderItem) }
            </div>
        );
    };

    renderItem = (item) => {
        return (
            <div className='dropdown-list-item' key={ item.id } onClick={ () => this.props.onDropdownItemClicked(item) }>
                {
                    item.isSelected
                        ? <img src='CheckedCircle.png' />
                        : <img src='EmptyCircle.png' />
                }
                <span className='dropdown-list-item-text'>{ item.text }</span>
            </div>
        );
    };

    render() {
        return (
            <div className='dropdown-wrapper'>
                { this.renderHeader() }
                { this.renderItemList() }
            </div >
        );
    }
}

Dropdown.propTypes = {
    title: PropTypes.string,
    items: PropTypes.arrayOf(PropTypes.shape({
        id: PropTypes.string,
        text: PropTypes.string,
        isSelected: PropTypes.bool,
    })),
    onDropdownItemClicked: PropTypes.func,
    sizeSmall: PropTypes.bool
};

export default onClickOutside(Dropdown);