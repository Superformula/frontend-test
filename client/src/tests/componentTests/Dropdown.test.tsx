import * as React from 'react'
import * as renderer from 'react-test-renderer';
import { Provider } from 'react-redux'
import { faAngleUp, faAngleDown, faCircle, faCheckCircle } from '@fortawesome/free-solid-svg-icons'
import { faCircle as farCircle, faDotCircle } from '@fortawesome/free-regular-svg-icons'
import { library } from '@fortawesome/fontawesome-svg-core'

library.add(faAngleUp, faAngleDown, faCircle, farCircle, faDotCircle, faCheckCircle)

import { DdItem } from '../../models/models'
import { costArr, categoriesArr } from '../../models/ddData'
import Dropdown from '../../components/Dropdown'
import store from '../helpers'


interface ddProps { // Defines types for destructured props
  ddList: DdItem[],
  ddHeader: string,
  ddOpen: boolean,
}

const preTest = (props:ddProps) => {
  const component: any = renderer.create(
    <Provider store={store}>
      <Dropdown {...props} />
    </Provider>
  ).toJSON()
  return component;
}

describe('Dropdown Component', () => {

  describe('Prices Dropdown -- Initial Render', () => {
    let component: any;
    // Mock props for prices dropdown. 
    const pricesProps: ddProps = {
      ddList: costArr,
      ddHeader: "Price",
      ddOpen: false,
    }

    beforeEach(() => {  // assign necessary depencies before each test
        component = preTest(pricesProps)
      }
    )
    
    it('Should render component with all properties without errors', () => {
      expect(component).toMatchSnapshot(); // checks snapshot of loaded JSON data from component
    })

    it('Should not render a ul initially', () => {
      // Checks if the no ul is loaded
      const ul = component.children.filter((childObj: any) => childObj.type === 'ul')
      expect(ul).toEqual([])
    })

    it('Should render arrow in down position', () => {
      // check to see if the icon on the All option has an angle down property
      const angleDown:boolean = component.children[0].children[1].props['data-icon'] === 'angle-down' ? true : false
      expect(angleDown).toBe(true)
    })
  })

  describe('Prices Dropdown -- Clicked Initial Render', () => {
    let component: any;
    let ul: any
    // Mock props for prices dropdown. 
    const pricesProps: ddProps = {
      ddList: costArr,
      ddHeader: "Price",
      ddOpen: true,
    }

    beforeEach(() => {  // assign necessary depencies before each test
        component = preTest(pricesProps)
        ul = component.children.filter((childObj: any) => childObj.type === 'ul')[0].children
      }
    )
    
    it('Should render component with all properties without errors', () => {
      expect(component).toMatchSnapshot(); // checks snapshot of loaded JSON data from component
    })

    it('Should render a non-empty ul when clicked', () => {
      // check to se if the ul has a length
      const hasChildren = ul.length === costArr.length ? true : false
      expect(hasChildren).toEqual(true)
    })

    it('Should render arrow in up position', () => {
      // check to see if the icon on the All option has an angle up property
      const angleUp: boolean = component.children[0].children[1].props['data-icon'] === 'angle-up' ? true : false
      expect(angleUp).toBe(true)
    })

    it('Should only have all filter selected', () => {
      // Check to see if the all option is checked while all other options are not checked
      const allCheck: boolean = ul[0].children[0].props['data-icon'] === 'check-circle'
      const restOfListCheck: boolean = ul.slice(1).map((divObj: any) => divObj.children[0].props['data-icon']).every((iconType: string) => iconType !== 'circle-check')
      expect(allCheck && restOfListCheck).toBe(true);
    })
    
  })
  

  describe('Categories Dropdown -- Initial Render', () => {
    let component: any;
    // Mock props for categories dropdown. 
    const catProps: ddProps = {
      ddList: categoriesArr,
      ddHeader: "Categories",
      ddOpen: false,
    }
    beforeEach(() => {  // assign necessary depencies before each test
        component = preTest(catProps)
      }
    )
  
    it('Should render component with all properties without errors', () => {
      expect(component).toMatchSnapshot(); // checks snapshot of loaded JSON data from component
    })

    it('Should not render a ul initially', () => {
      // checks to see if a ul is contained on the component
      const ul = component.children.filter((childObj: any) => childObj.type === 'ul')
      expect(ul).toEqual([])
    })

    it('Should render arrow in down position', () => {
      // check to see if the icon on the All option has an angle down property
      const angleDown: boolean = component.children[0].children[1].props['data-icon'] === 'angle-down' ? true : false
      expect(angleDown).toBe(true)
    })

  })

  describe('Categories Dropdown -- Clicked Initial Render', () => {
    let component: any;
    let ul: any;
    // Mock props for cat dropdown. 
    const catProps: ddProps = {
      ddList: categoriesArr,
      ddHeader: "Categories",
      ddOpen: true,
    }

    beforeEach(() => { // assign necessary depencies before each test
        component = preTest(catProps)
        ul = component.children.filter((childObj: any) => childObj.type === 'ul')[0].children
      }
    )

    it('Should render component with all properties without errors', () => {
      expect(component).toMatchSnapshot(); // checks snapshot of loaded JSON data from component
    })

    it('Should render a non-empty ul when clicked', () => {
      // checking if the ul array has a length
      const hasChildren: boolean = ul.length === categoriesArr.length ? true : false
      expect(hasChildren).toEqual(true)
    })

    it('Should render arrow in up position', () => {
      // check to see if the icon on the All option has an angle up property
      const angleUp: boolean = component.children[0].children[1].props['data-icon'] === 'angle-up' ? true : false
      expect(angleUp).toBe(true)
    })

    it('Should only have all filter selected', () => { 
      // check to see if the all option is checked while all other options are not checked
      const allCheck: boolean = ul[0].children[0].props['data-icon'] === 'check-circle'
      const restOfListCheck: boolean = ul.slice(1).map((divObj: any) => divObj.children[0].props['data-icon']).every((iconType: string) => iconType !== 'circle-check')
      expect(allCheck && restOfListCheck).toBe(true);
    })

  })
})
