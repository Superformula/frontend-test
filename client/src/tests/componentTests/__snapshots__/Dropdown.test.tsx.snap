// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown Component Categories Dropdown -- Clicked Initial Render Should render component with all properties without errors 1`] = `
<div
  className="dd-header"
>
  <div
    className="dd-header-title"
    onClick={[Function]}
  >
    Categories  
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-angle-up fa-w-10 "
      color="#C8C8C8"
      data-icon="angle-up"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 320 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M177 159.7l136 136c9.4 9.4 9.4 24.6 0 33.9l-22.6 22.6c-9.4 9.4-24.6 9.4-33.9 0L160 255.9l-96.4 96.4c-9.4 9.4-24.6 9.4-33.9 0L7 329.7c-9.4-9.4-9.4-24.6 0-33.9l136-136c9.4-9.5 24.6-9.5 34-.1z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
  </div>
  <ul
    className="dd-list"
  >
    <div
      className="dd-list-item"
      onClick={[Function]}
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-check-circle fa-w-16 "
        color="#002B56"
        data-icon="check-circle"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M504 256c0 136.967-111.033 248-248 248S8 392.967 8 256 119.033 8 256 8s248 111.033 248 248zM227.314 387.314l184-184c6.248-6.248 6.248-16.379 0-22.627l-22.627-22.627c-6.248-6.249-16.379-6.249-22.628 0L216 308.118l-70.059-70.059c-6.248-6.248-16.379-6.248-22.628 0l-22.627 22.627c-6.248 6.248-6.248 16.379 0 22.627l104 104c6.249 6.249 16.379 6.249 22.628.001z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
        All
    </div>
    <div
      className="dd-list-item"
      onClick={[Function]}
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-circle fa-w-16 "
        color="#002B56"
        data-icon="circle"
        data-prefix="far"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm0 448c-110.5 0-200-89.5-200-200S145.5 56 256 56s200 89.5 200 200-89.5 200-200 200z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
        Italian
    </div>
    <div
      className="dd-list-item"
      onClick={[Function]}
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-circle fa-w-16 "
        color="#002B56"
        data-icon="circle"
        data-prefix="far"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm0 448c-110.5 0-200-89.5-200-200S145.5 56 256 56s200 89.5 200 200-89.5 200-200 200z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
        Seafood
    </div>
    <div
      className="dd-list-item"
      onClick={[Function]}
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-circle fa-w-16 "
        color="#002B56"
        data-icon="circle"
        data-prefix="far"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm0 448c-110.5 0-200-89.5-200-200S145.5 56 256 56s200 89.5 200 200-89.5 200-200 200z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
        Steakhouses
    </div>
    <div
      className="dd-list-item"
      onClick={[Function]}
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-circle fa-w-16 "
        color="#002B56"
        data-icon="circle"
        data-prefix="far"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm0 448c-110.5 0-200-89.5-200-200S145.5 56 256 56s200 89.5 200 200-89.5 200-200 200z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
        Japanese
    </div>
    <div
      className="dd-list-item"
      onClick={[Function]}
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-circle fa-w-16 "
        color="#002B56"
        data-icon="circle"
        data-prefix="far"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm0 448c-110.5 0-200-89.5-200-200S145.5 56 256 56s200 89.5 200 200-89.5 200-200 200z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
        American
    </div>
    <div
      className="dd-list-item"
      onClick={[Function]}
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-circle fa-w-16 "
        color="#002B56"
        data-icon="circle"
        data-prefix="far"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm0 448c-110.5 0-200-89.5-200-200S145.5 56 256 56s200 89.5 200 200-89.5 200-200 200z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
        Mexican
    </div>
    <div
      className="dd-list-item"
      onClick={[Function]}
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-circle fa-w-16 "
        color="#002B56"
        data-icon="circle"
        data-prefix="far"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm0 448c-110.5 0-200-89.5-200-200S145.5 56 256 56s200 89.5 200 200-89.5 200-200 200z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
        Thai
    </div>
  </ul>
</div>
`;

exports[`Dropdown Component Categories Dropdown -- Initial Render Should render component with all properties without errors 1`] = `
<div
  className="dd-header"
>
  <div
    className="dd-header-title"
    onClick={[Function]}
  >
    Categories  
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-angle-down fa-w-10 "
      color="#C8C8C8"
      data-icon="angle-down"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 320 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M143 352.3L7 216.3c-9.4-9.4-9.4-24.6 0-33.9l22.6-22.6c9.4-9.4 24.6-9.4 33.9 0l96.4 96.4 96.4-96.4c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9l-136 136c-9.2 9.4-24.4 9.4-33.8 0z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
  </div>
</div>
`;

exports[`Dropdown Component Prices Dropdown -- Clicked Initial Render Should render component with all properties without errors 1`] = `
<div
  className="dd-header"
>
  <div
    className="dd-header-title"
    onClick={[Function]}
  >
    Price  
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-angle-up fa-w-10 "
      color="#C8C8C8"
      data-icon="angle-up"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 320 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M177 159.7l136 136c9.4 9.4 9.4 24.6 0 33.9l-22.6 22.6c-9.4 9.4-24.6 9.4-33.9 0L160 255.9l-96.4 96.4c-9.4 9.4-24.6 9.4-33.9 0L7 329.7c-9.4-9.4-9.4-24.6 0-33.9l136-136c9.4-9.5 24.6-9.5 34-.1z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
  </div>
  <ul
    className="dd-list"
  >
    <div
      className="dd-list-item"
      onClick={[Function]}
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-check-circle fa-w-16 "
        color="#002B56"
        data-icon="check-circle"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M504 256c0 136.967-111.033 248-248 248S8 392.967 8 256 119.033 8 256 8s248 111.033 248 248zM227.314 387.314l184-184c6.248-6.248 6.248-16.379 0-22.627l-22.627-22.627c-6.248-6.249-16.379-6.249-22.628 0L216 308.118l-70.059-70.059c-6.248-6.248-16.379-6.248-22.628 0l-22.627 22.627c-6.248 6.248-6.248 16.379 0 22.627l104 104c6.249 6.249 16.379 6.249 22.628.001z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
        All
    </div>
    <div
      className="dd-list-item"
      onClick={[Function]}
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-circle fa-w-16 "
        color="#002B56"
        data-icon="circle"
        data-prefix="far"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm0 448c-110.5 0-200-89.5-200-200S145.5 56 256 56s200 89.5 200 200-89.5 200-200 200z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
        $
    </div>
    <div
      className="dd-list-item"
      onClick={[Function]}
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-circle fa-w-16 "
        color="#002B56"
        data-icon="circle"
        data-prefix="far"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm0 448c-110.5 0-200-89.5-200-200S145.5 56 256 56s200 89.5 200 200-89.5 200-200 200z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
        $$
    </div>
    <div
      className="dd-list-item"
      onClick={[Function]}
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-circle fa-w-16 "
        color="#002B56"
        data-icon="circle"
        data-prefix="far"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm0 448c-110.5 0-200-89.5-200-200S145.5 56 256 56s200 89.5 200 200-89.5 200-200 200z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
        $$$
    </div>
    <div
      className="dd-list-item"
      onClick={[Function]}
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-circle fa-w-16 "
        color="#002B56"
        data-icon="circle"
        data-prefix="far"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm0 448c-110.5 0-200-89.5-200-200S145.5 56 256 56s200 89.5 200 200-89.5 200-200 200z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
        $$$$
    </div>
  </ul>
</div>
`;

exports[`Dropdown Component Prices Dropdown -- Initial Render Should render component with all properties without errors 1`] = `
<div
  className="dd-header"
>
  <div
    className="dd-header-title"
    onClick={[Function]}
  >
    Price  
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-angle-down fa-w-10 "
      color="#C8C8C8"
      data-icon="angle-down"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 320 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M143 352.3L7 216.3c-9.4-9.4-9.4-24.6 0-33.9l22.6-22.6c9.4-9.4 24.6-9.4 33.9 0l96.4 96.4 96.4-96.4c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9l-136 136c-9.2 9.4-24.4 9.4-33.8 0z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
  </div>
</div>
`;
